### PR TITLE
Fixes to PETSc examples [stefanozampini/fix-example]

### PR DESCRIPTION
**Pull request by stefanozampini**
This just adds few tweaks to the PETSc examples

### CODE REVIEW

File: linalg/petsc.cpp [hunk 1]
```diff
@@ -834,7 +834,7 @@ void PetscParMatrix::ConvertOperator(MPI_Comm comm, const Operator &op, Mat* A,
                if (tid == PETSC_MATIS && needl2l)
                {
                   PetscContainer c;
-                  ierr = PetscObjectQuery((PetscObject)mats[i*nc+j],"__mfem_l2l",
+                  ierr = PetscObjectQuery((PetscObject)mats[i*nc+j],"_MatIS_PtAP_l2l",
```
Comment (v-dobrev) [inline linalg/petsc.cpp]: I'm not sure what the role of the modified string is: does this change mean that MFEM now requires PETSc 3.9.0 or newer?

**Comment by stefanozampini** [comment]
No need to use a newer version . This is for newer versions of petsc to
take advantage of mfem, not the other way around

Il Sab 16 Giu 2018, 02:07 Veselin Dobrev <notifications@github.com> ha
scritto:

**Comment by v-dobrev** [comment]
Hey Stefano, I tried running the PETSc tests in a build using PETSc 3.8.3, and the first 2 test runs for ex5p failed. Is this due to the old PETSc version? If that is the case, we should skip those tests when using older versions (I assume versions < 3.9.0?).

**Comment by v-dobrev** [comment]
Or is this issue showing up just because of the modified `rc_*` files?

**Comment by stefanozampini** [comment]
which tests are failing? the one for ex5p need PETSc compiled with support with mumps and umfpack according to the rc file. The change for 3.9 wrt older versions is that the option -pc_factor_mat_solver_package is now -pc_factor_mat_solver_type as explained in the rc files.

**Comment by v-dobrev** [comment]
The failing tests are the ones with arguments defined by `EX5_BDDC_LB_ARGS` and `EX5_BDDC_GB_ARGS` in the makefile. In the PETSc build (v3.8.3) I used, mumps and umfpack were enabled - that is not the issue. The question is if these failures are due just to the modified (for PETSc v3.9) `rc_*` files, or if these two tests use some functionality that is new in PETSc 3.9 and simply fixing the `rc_*` files will not be sufficient?

**Comment by stefanozampini** [comment]
@v-dobrev the examples should be fixed now. Can you check on your side?

**Comment by v-dobrev** [comment]
@stefanozampini, this works great, and there is no need to change the `rc_*` files when using different versions of PETSc - I did not realize you can set both the old and the new options. Thanks!

**Pull request merged**